src/config/schema.ts

```diff
@@ -0,0 +1,360 @@
+      env: 'MAIL_SOCKET_TIMEOUT',
+    },
+  },
+  awsConfig: {
+    region: {
+      doc: 'Region that S3 bucket is located in',
+      format: Object.keys(awsInfo.data.regions),
+      default: 'ap-southeast-1',
+      env: 'AWS_REGION',
+    },
+    customCloudWatchGroup: {
+      doc:
+        'Name of CloudWatch log group to store short-term logs. Log streams are separated by date.',
+      format: String,
+      default: '',
+      env: 'CUSTOM_CLOUDWATCH_LOG_GROUP',
+    },
+  },
+  core: {
+    port: {
+      doc: 'Application Port',
+      format: 'port',
+      default: 5000,
+      env: 'PORT',
+    },
+    otpLifeSpan: {
+      doc:
+        'OTP Life Span for Login. (Should be in miliseconds, e.g. 1000 * 60 * 15 = 15 mins)',
+      format: 'int',
+      default: 900000,
+      env: 'OTP_LIFE_SPAN',
+    },
+    submissionsTopUp: {
+      doc: 'Number of submissions to top up submissions statistic by',
+      format: 'int',
+      default: 0,
+      env: 'SUBMISSIONS_TOP_UP',
+    },
+    nodeEnv: {
+      doc: 'Express environment mode',
+      format: [Environment.Prod, Environment.Dev, Environment.Test],
+      default: Environment.Prod,
+      env: 'NODE_ENV',
+    },
+  },
+}
+
+export const prodOnlyVarsSchema: Schema<IProdOnlyVarsSchema> = {
+  port: {
+    doc: 'SMTP port number',
+    format: 'port',
+    default: null,
+    env: 'SES_PORT',
+  },
+  host: {
+    doc: 'SMTP hostname',
+    format: String,
+    default: null,
+    env: 'SES_HOST',
+  },
+  user: {
+    doc: 'SMTP username',
+    format: String,
+    default: null,
+    env: 'SES_USER',
+  },
+  pass: {
+    doc: 'SMTP password',
+    format: String,
+    default: null,
+    env: 'SES_PASS',
+    sensitive: true,
+  },
+  dbHost: {
+    doc: 'Database URI',
+    format: (val) => {
+      // Will throw error if scheme and hosts are not present
+      const uriObject = mongodbUri.parse(val)
+      /*
+        e.g. mongodb://database:27017/formsg will be parsed into:
+        {
+          scheme: 'mongodb',
+          database: 'formsg',
+          hosts: [ { host: 'database', port: 27017 } ]
+        }
+        e.g. https://form.gov.sg will be parsed into: 
+        { 
+          scheme: 'https', 
+          hosts: [ { host: 'form.gov.sg' } ] 
+        }
+      */
+      if (uriObject.scheme !== 'mongodb') {
+        throw new Error('Scheme must be mongodb')
+      }
+      if (isNil(uriObject.database)) {
+        throw new Error('Database must be specified')
+      }
+    },
+    default: null,
+    env: 'DB_HOST',
+    sensitive: true,
+  },
+}
+
+export const loadS3BucketUrlSchema = ({
+  isDev,
+  region,
+}: {
+  isDev: boolean
+  region: string
+}): Schema<IBucketUrlSchema> => {
+  return {
+    endPoint: {
+      doc: 'Endpoint for S3 buckets',
+      format: (val) =>
+        validateBucketUrl(val, { isDev, hasTrailingSlash: false, region }),
+      default: 'https://s3.ap-southeast-1.amazonaws.com', // NOTE NO TRAILING / AT THE END OF THIS URL!
```

Comment:
hm, I wonder if we should just make this default `null` so it always must be explicit?

Reply:
There is little reason for this env var to change? Any reason why it cannot have a default?

Reply:
I guess I was just imagining it might be annoying on the dev environment - let's say we merge this change, but someone doesn't add `AWS_ENDPOINT` to their dev environment. Then all Localstack stuff will break silently because the dev environment won't throw an error (because of this default), but requests to this endpoint will obviously fail

Reply:
but I guess that concern is fixed by docker-compose so I'm ok with this as it is